code/game/machinery/newscaster/newscaster_machine.dm

```diff
@@ -0,0 +1,551 @@
+				return
+			else
+				attach_photo(usr)
+				if(current_image)
+					balloon_alert(usr, "Photo selected.")
+				else
+					balloon_alert(usr, "No photo identified.")
+
+		if("startCreateChannel")
+			//This first block checks for pre-existing reasons to prevent you from making a new channel, like being censored, or if you have a channel already.
+			var/list/existing_authors = list()
+			for(var/datum/newscaster/feed_channel/FeedC in GLOB.news_network.network_channels)
+				if(FeedC.authorCensor)
+					existing_authors += GLOB.news_network.redactedText
+				else
+					existing_authors += FeedC.author
+			if(current_user?.account_holder == ("Unknown" || null) || (current_user.account_holder in existing_authors))
+				tgui_alert(usr, "ERROR: User cannot be found or already has an owned feed channel.", list("Okay"))
+				return
+			creating_channel = TRUE
+		if("channelName")
+			channel_name = (params["channeltext"])
+		if("channelDesc")
+			channel_desc = (params["channeldesc"])
+
+		if("createChannel")
+			if(!channel_name)
+				return
+			for(var/datum/newscaster/feed_channel/FeedC in GLOB.news_network.network_channels)
+				if(FeedC.channel_name == channel_name)
+					tgui_alert(usr, "ERROR: Feed Channel with that name already exists on the Network.", list("Okay"))
+					return
+			if(!channel_desc)
+				return
+			var/locked = (params["publicmode"])
+			var/choice = tgui_alert(usr, "Please confirm Feed channel creation","Network Channel Handler", list("Confirm","Cancel"))
+			if(choice=="Confirm")
+				GLOB.news_network.CreateFeedChannel(channel_name, current_user.account_holder, channel_desc , locked)
+				SSblackbox.record_feedback("text", "newscaster_channels", 1, "[channel_name]")
+			creating_channel = FALSE
+
+		if("storyCensor")
+			if (!params["secure"])
+				say("Clearance not found.")
+				return
+			var/questionable_message = params["messageID"]
+			for(var/datum/newscaster/feed_message/mess in current_channel.messages)
+				if(mess.message_ID == questionable_message)
+					mess.toggleCensorBody()
+					break
+
+		if("authorCensor")
+			if (!params["secure"])
+				say("Clearance not found.")
+				return
+			var/questionable_message = params["messageID"]
+			for(var/datum/newscaster/feed_message/mess in current_channel.messages)
+				if(mess.message_ID == questionable_message)
+					mess.toggleCensorAuthor()
+					break
+
+		if("channelDNotice")
+			if (!params["secure"])
+				say("Clearance not found.")
+				return
+			var/proto_chan = (params["channel"])
+			for(var/datum/newscaster/feed_channel/potential_channel in GLOB.news_network.network_channels)
+				if(proto_chan == potential_channel.channel_ID)
+					current_channel = potential_channel
+					break
+			current_channel.toggleCensorDclass()
+
+		if("printNewspaper")
+			if(paper_remaining <= 0)
+				balloon_alert_to_viewers("Unit out of Paper!")
+				return
+			print_paper()
+
+		//**************************
+		//	  Bounty Board Acts
+		//**************************
```

Comment:
Not useful, remove it or just make it `// Bounty board acts`.

Honestly, this proc is *huge*, and should likely be split into multiple anyway.